libs/features/mas/src/utils.js

```diff
@@ -81,3 +81,35 @@ export function discoverService(getConfig, { once = false } = {}) {
 export function getService() {
   return document.getElementsByTagName(MAS_COMMERCE_SERVICE)?.[0];
 }
+
+function historyPushState(queryParams) {
+    if (!queryParams || !window.history.pushState) return;
+    const newURL = new URL(window.location.href);
+    newURL.search = `?${queryParams}`;
+    window.history.pushState({ path: newURL.href }, '', newURL.href);    
+}
+
+export function updateHash(key, value) {
+    const hash = new URLSearchParams(window.location.hash.slice(1));
+    hash.set(key, value);
+    window.location.hash = hash.toString();
+}
+
+/**
+ * Convert the query params to hash
+ * @param {string[]} keys - The keys to convert to hash
+ */
+export function paramsToHash(keys = []) {
```

Comment:
a unit test for this function would be good.

Reply:
> queryLabel doesn't make sense though, just node.queryParam or node.queryParameter?
`queryLabel` is the name we put on the model in odin side.
I used the same name to be matching everywhere.